lib/msal-browser/src/math/GuidGenerator.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { MathUtils } from "../utils/MathUtils";
+
+export class GuidGenerator {
+
+    private cryptoObj: Crypto;
+
+    constructor(cryptoObj: Crypto) {
+        this.cryptoObj = cryptoObj;
+    }
+
+    /*
+     * RFC4122: The version 4 UUID is meant for generating UUIDs from truly-random or
+     * pseudo-random numbers.
+     * The algorithm is as follows:
+     *     Set the two most significant bits (bits 6 and 7) of the
+     *        clock_seq_hi_and_reserved to zero and one, respectively.
+     *     Set the four most significant bits (bits 12 through 15) of the
+     *        time_hi_and_version field to the 4-bit version number from
+     *        Section 4.1.3. Version4
+     *     Set all the other bits to randomly (or pseudo-randomly) chosen
+     *     values.
+     * UUID                   = time-low "-" time-mid "-"time-high-and-version "-"clock-seq-reserved and low(2hexOctet)"-" node
+     * time-low               = 4hexOctet
+     * time-mid               = 2hexOctet
+     * time-high-and-version  = 2hexOctet
+     * clock-seq-and-reserved = hexOctet:
+     * clock-seq-low          = hexOctet
+     * node                   = 6hexOctet
+     * Format: xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx
+     * y could be 1000, 1001, 1010, 1011 since most significant two bits needs to be 10
+     * y values are 8, 9, A, B
+     */
+    generateGuid(): string {
```

Comment:
Is there not a library that does all this work for us?

Reply:
https://www.npmjs.com/package/uuidv4

Reply:
This is currently how we create guids today. I am open to using external libraries, but in the past we have implemented similar functions ourselves due to issues in downstream libraries.

Reply:
Although this library looks pretty stable, so I have no objections to updating usage.

Reply:
Depending on someone else would mean we are depended on the js/ts version support/ polyfill / es6 support etc. My recommendation would be to stick to our implementation as this is straightforward.

Reply:
I'm fine with leaving this how it is if you all prefer our own implementation--I generally just prefer using libraries to reduce the # of lines of code we own, like in this case.